File: L2jOrion_Datapack/game/data/scripts/ai/group_template/frozen.py
import sys
from l2jorion.game.ai import CtrlIntention
from l2jorion.game.model.quest import State
from l2jorion.game.model.quest import QuestState
from l2jorion.game.model.quest.jython import QuestJython as JQuest
from l2jorion.util.random import Rnd

FrozenId = {
    #Head Mob ID [Minion ID, Minion count]
    22094:[22093,5],
    22088:[22087,5]
}

class frozen(JQuest) :

    def __init__(self,id,name,descr):
        self.AlwaysSpawn = False
        JQuest.__init__(self,id,name,descr)

    def onAttack(self,npc,player,isPet,damage):
        npcId = npc.getNpcId()
        NewMob,mob_count = FrozenId[npcId]
        if FrozenId.has_key(npcId) :
            if self.AlwaysSpawn == True:
                return
            elif self.AlwaysSpawn == False and player.isAttackingNow():
                count = 0
                self.AlwaysSpawn = True
                while count < mob_count:
                    New = self.addSpawn(NewMob,npc.getX(), npc.getY(), npc.getZ(),npc.getHeading(),True,0)
                    New.addDamageHate(player,0,999)
                    New.getAI().setIntention(CtrlIntention.AI_INTENTION_ATTACK, player)
                    count = count+1
        return 

    def onKill (self,npc,player,isPet):
        npcId = npc.getNpcId()
        NewMob,mob_count = FrozenId[npcId]
        if FrozenId.has_key(npcId) :
            self.AlwaysSpawn = False
        return

QUEST = frozen(-1,"frozen","ai")

CREATED = State('Start', QUEST)
QUEST.setInitialState(CREATED)

for i in FrozenId.keys() :
   QUEST.addAttackId(i)

for j in FrozenId.keys() :
   QUEST.addKillId(j)

File: L2jOrion_SourceCode/src/l2jorion/game/ai/additional/Frozen.java
package l2jorion.game.ai.additional;

import java.util.HashSet;
import java.util.Map;
import java.util.Set;
import java.util.concurrent.ConcurrentHashMap;

import l2jorion.game.ai.CtrlIntention;
import l2jorion.game.model.L2Attackable;
import l2jorion.game.model.L2Skill;
import l2jorion.game.model.actor.instance.L2NpcInstance;
import l2jorion.game.model.actor.instance.L2PcInstance;
import l2jorion.game.model.quest.Quest;
import l2jorion.game.util.Util;
import l2jorion.util.random.Rnd;

public class Frozen extends Quest implements Runnable
{
	private static int HasSpawned;
	private static Set<Integer> TrackingSet = new HashSet<>();
	
	static final int[] mobs =
	{
		22094,
		22088
	};
	
	private static final Map<Integer, Integer[]> MINIONS = new ConcurrentHashMap<>();
	
	static
	{
		MINIONS.put(22094, new Integer[]
		{
			22093,
			22093,
			22093,
			22093,
			22093
		});
		MINIONS.put(22088, new Integer[]
		{
			22087,
			22087,
			22087,
			22087,
			22087
		});
	}
	
	public Frozen(int questId, String name, String descr)
	{
		super(questId, name, descr);
		registerMobs(mobs, QuestEventType.ON_SKILL_USE);
	}
	
	@Override
	public String onSkillUse(L2NpcInstance npc, L2PcInstance caster, L2Skill skill)
	{
		int npcId = npc.getNpcId();
		int npcObjId = npc.getObjectId();
		
		if (Util.contains(mobs, npcId))
		{
			if (skill.isMagic())
			{
				return null;
			}
			
			if (!TrackingSet.contains(npcObjId))
			{
				TrackingSet.add(npcObjId);
				HasSpawned = npcObjId;
			}
			
			if (HasSpawned == npcObjId)
			{
				HasSpawned = 0;
				Integer[] minions = MINIONS.get(npcId);
				for (Integer minion : minions)
				{
					L2Attackable newNpc = (L2Attackable) addSpawn(minion, (npc.getX() + Rnd.get(-150, 150)), (npc.getY() + Rnd.get(-150, 150)), npc.getZ(), 0, false, 0);
					newNpc.setRunning();
					newNpc.addDamageHate(caster, 0, 999);
					newNpc.getAI().setIntention(CtrlIntention.AI_INTENTION_ATTACK, caster);
				}
			}
		}
		return super.onSkillUse(npc, caster, skill);
	}
	
	@Override
	public void run()
	{
	}
}